build-system/pr-check/validator.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * This script runs validator tests.
+ * This is run during the CI stage = build; job = validator.
+ */
+
+const {determineBuildTargets} = require('./build-target');
+const {isTravisPushBuild} = require('../travis');
+const {startTimer, stopTimer, timedExecOrDie} = require('./utils');
+
+const FILENAME = 'validator.js';
```

Comment:
This job will be a no-op most of the time, and yet, there's the overhead of spinning up a VM to do nothing. 

One way to avoid this is to run a standalone `nodejs` script in `.travis.yml` to determine the build targets, and make the various jobs start up conditionally.

Definitely out of scope for this PR, but let's consider this as a future optimization, since it will save a whole lot of VM spin-ups in the long run.

Reply:
Agreed! Would like to utilize Travis's build conditions feature, so a job doesn't show as "succeeded" when it really should be "skipped" or hidden.